mautic/mautic
# retrigger gitpod image generation and update ddev to latest version

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

When troubleshooting a test failure, I noticed the gitpod `composer` version is still 1.10 (inside the ddev shell it's 2.5). The ddev version is also outdated.

The main cause is that gitpod caches the generated image, and it is only regenerated when the dockerfile changes.
Composer 2 support was added in gitpod on Feb 4, 2022, [see here](https://github.com/gitpod-io/workspace-images/pull/660), but the last change of the dockerfile was on Nov 9, 2021, [see here](https://github.com/mautic/mautic/pull/10500).

This PR ensures the latest `gitpod/workspace-full` is used (by regenerating the Mautic gitpod image), and ddev is up-to-date. Next to that, ddev is installed via the recommended way (see https://ddev.readthedocs.io/en/latest/users/install/ddev-installation/), and the phpmyadmin is added separately, as it's not part anymore of the ddev core

**before:**
```
gitpod /workspace/mautic $ composer -V
Deprecation Notice: Required parameter $path follows optional parameter $schema in /usr/share/php/JsonSchema/Constraints/UndefinedConstraint.php:62
Deprecation Notice: Required parameter $path follows optional parameter $schema in /usr/share/php/JsonSchema/Constraints/UndefinedConstraint.php:108
Deprecation Notice: Method ReflectionParameter::getClass() is deprecated in /usr/share/php/Composer/Repository/RepositoryManager.php:130
Deprecation Notice: Method ReflectionParameter::getClass() is deprecated in /usr/share/php/Composer/Repository/RepositoryManager.php:130
Deprecation Notice: Method ReflectionParameter::getClass() is deprecated in /usr/share/php/Composer/Repository/RepositoryManager.php:130
Deprecation Notice: Method ReflectionParameter::getClass() is deprecated in /usr/share/php/Composer/Repository/RepositoryManager.php:130
Deprecation Notice: Method ReflectionParameter::getClass() is deprecated in /usr/share/php/Composer/Repository/RepositoryManager.php:130
Composer 1.10.1 2020-03-13 20:34:27

gitpod /workspace/mautic $ ddev version
 ITEM            VALUE                                   
 DDEV version    v1.18.2                                 
 architecture    amd64                                   
 db              drud/ddev-dbserver-mariadb-10.3:v1.18.2 
 dba             phpmyadmin:5                            
 ddev-ssh-agent  drud/ddev-ssh-agent:v1.18.0             
 docker          20.10.8                                 
 docker-compose  v2.2.2                                  
 mutagen         0.12.0                                  
 os              linux                                   
 router          drud/ddev-router:v1.18.2                
 web             drud/ddev-webserver:v1.18.2             

```

**after:**
```
gitpod /workspace/mautic (gitpod-image-update) $ composer -V
Composer version 2.5.5 2023-03-21 11:50:05

gitpod /workspace/mautic (gitpod-image-update) $ ddev version
 ITEM             VALUE                                   
 DDEV version     v1.21.6                                 
 architecture     amd64                                   
 db               drud/ddev-dbserver-mariadb-10.4:v1.21.5 
 dba              phpmyadmin:5                            
 ddev-ssh-agent   drud/ddev-ssh-agent:v1.21.5             
 docker           23.0.3                                  
 docker-compose   v2.15.1                                 
 docker-platform  mollux-mautic-thpbyh2qnfo               
 mutagen          0.16.0                                  
 os               linux                                   
 router           drud/ddev-router:v1.21.5                
 web              drud/ddev-webserver:v1.21.5                       
 
```



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. verify that you can login into Mautic (which means DDEV is working)
3. verify phpmyadmin (port 8036) is working
4. check the ddev version: `ddev version`
5. check the composer version: `composer -V`
